### ISSUE TITLE
Exception when opening plant project with invalid video url

### ISSUE THREAD

**Issue by norbertschuler** [Plant-for-the-Planet-org/treecounter-app#906]
When clicking on the project "Königswald Ontananga" the iOS simulator shows an error:

![Simulator Screen Shot - iPhone X - 2019-06-12 at 09 50 07](https://user-images.githubusercontent.com/1532418/59332843-78064b80-8cf7-11e9-91e4-acbb9c7675ba.png)

```
JSON value '(
    "https://www.youtube.com/channel/UCPE2SUtnxjr4qU-O2VuBkxQ"
)' of type NSMutableArray cannot be converted to NSString

+[RCTConvert NSString:]
    RCTConvert.m:56
__49-[RCTComponentData createPropBlock:isShadowView:]_block_invoke.95
__49-[RCTComponentData createPropBlock:isShadowView:]_block_invoke_2.96
__49-[RCTComponentData propBlockForKey:isShadowView:]_block_invoke_2
RCTPerformBlockWithLogFunction
RCTPerformBlockWithLogPrefix
__49-[RCTComponentData propBlockForKey:isShadowView:]_block_invoke
__37-[RCTComponentData setProps:forView:]_block_invoke
-[__NSDictionaryM enumerateKeysAndObjectsWithOptions:usingBlock:]
-[RCTComponentData setProps:forView:]
__50-[RCTUIManager createView:viewName:rootTag:props:]_block_invoke.407
__44-[RCTUIManager flushUIBlocksWithCompletion:]_block_invoke
__44-[RCTUIManager flushUIBlocksWithCompletion:]_block_invoke.463
__RCTExecuteOnMainQueue_block_invoke
_dispatch_call_block_and_release
_dispatch_client_callout
_dispatch_main_queue_callback_4CF
__CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__
__CFRunLoopRun
CFRunLoopRunSpecific
GSEventRunModal
UIApplicationMain
main
start
```

This exception does not seem to be catched from Bugsnag, so I post it here.

**Comment by norbertschuler** [comment]
I have found another example: When you open a project which has an invalid YouTube link. 

On iOS it just do not display the video as following:

![Simulator Screen Shot - iPhone X - 2019-07-21 at 12 07 36](https://user-images.githubusercontent.com/1532418/61587283-83c81600-abb1-11e9-80b2-57816dbc2041.png)
![Simulator Screen Shot - iPhone X - 2019-07-21 at 12 07 45](https://user-images.githubusercontent.com/1532418/61587284-87f43380-abb1-11e9-8044-cf7a845e94e1.png)
![Simulator Screen Shot - iPhone X - 2019-07-21 at 12 07 50](https://user-images.githubusercontent.com/1532418/61587286-8aef2400-abb1-11e9-8dac-50530632fecd.png)

On Android an invalid YouTube-Link can lead to a crash:
```
Error while updating property 'videoId' of a view managed by: ReactYouTube

Value for videoId cannot be cast from ReadableNativeArray to String
updateViewProp
    ViewManagersPropertyCache.java:95
setProperty
    ViewManagerPropertyUpdater.java:132
updateProps
    ViewManagerPropertyUpdater.java:51
updateProperties
    ViewManager.java:37
createView
    NativeViewHierarchyManager.java:265
execute
    UIViewOperationQueue.java:200
dispatchPendingNonBatchedOperations
    UIViewOperationQueue.java:1109
doFrameGuarded
    UIViewOperationQueue.java:1080
doFrame
    GuardedFrameCallback.java:29
doFrame
    ReactChoreographer.java:166
doFrame
    ChoreographerCompat.java:84
run
    Choreographer.java:947
doCallbacks
    Choreographer.java:761
doFrame
    Choreographer.java:693
run
    Choreographer.java:935
handleCallback
    Handler.java:873
dispatchMessage
    Handler.java:99
loop
    Looper.java:193
main
    ActivityThread.java:6669
invoke
    Method.java
run
    RuntimeInit.java:493
main
    ZygoteInit.java:858
```

It would be nice if the Android app wouldn't crash.

**Comment by norbertschuler** [comment]
Here is the issue reported from Bugsnag: #947

**Comment by rarawal** [comment]
@harshkurra : Please look into this.

**Comment by hkurra** [comment]
@norbertschuler front end requires Video ID instead of Video URL, so we are parsing URL to get Video ID, I have fixed the crash on android under Issue #947 
Now, earlier if URL did not have any Video ID then VideoContainer were passing URL as Video Id assuming it is video id but now I have provided one more prop `videoId` if the consumer wants to provide VideoID
